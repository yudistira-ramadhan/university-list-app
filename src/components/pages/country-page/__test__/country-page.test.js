import React from 'react'
import ReactDOM from 'react-dom'
import CountryPage from '../country-page'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<CountryPage/>, div)
})

it('renders CountryPage correctly', () => {
  const { getByTestId } = render(<CountryPage />)
  expect(getByTestId('country-page')).toBeTruthy()
})

describe('input value', ()=> {
  it('updates on change', ()=> {

    const { getByPlaceholderText } = render(<CountryPage/>)

    const searchInput = getByPlaceholderText('Input country name...')

    fireEvent.change(searchInput, { target: { value: 'test' } })

    expect(searchInput.value).toBe('test')
  })
})

it('matches snapshot', () => {
  const tree = renderer.create(<CountryPage />).toJSON()
  expect(tree).toMatchSnapshot()
})