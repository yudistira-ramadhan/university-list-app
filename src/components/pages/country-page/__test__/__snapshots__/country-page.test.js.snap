// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="country-page"
>
  <div
    className="text-center jumbotron"
  >
    <h2
      className="font-weight-bolder"
    >
      University List by Country
    </h2>
    <hr
      className="my-4"
    />
    <p
      className="lead"
    >
      You can see the list of University from the specific Country by input the Country name.
    </p>
    <div
      className="text-center"
      data-testid="search-field"
    >
      <div
        className="d-flex"
      >
        <input
          className="mr-2 form-control"
          name="country"
          onChange={[Function]}
          placeholder="Input country name..."
          type="text"
        />
        <button
          aria-label={null}
          className="btn btn-outline-primary"
          data-testid="search-button"
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</div>
`;
